### PR TITLE
Gpl 769 refactor cherrypicked helper

### PR DESCRIPTION
**Pull request by emrojo**
Closes #

Changes proposed in this pull request:

*
*
* ...

### CODE REVIEW

File: .release_version [hunk 1]
```diff
@@ -1 +1 @@
-1.9.3
+1.9.4
```
Comment (pjvv) [inline .release_version:1]: Is this a bugfix?

File: crawler/helpers/cherrypicked_samples.py [hunk 2]
```diff
@@ -0,0 +1,199 @@
...
+from crawler.types import Config, SampleDoc
+
+logger = logging.getLogger(__name__)
+
+
+def extract_required_cp_info(samples: List[SampleDoc]) -> Tuple[Set[str], Set[str]]:
+    root_sample_ids = set()
+    plate_barcodes = set()
+
+    for sample in samples:
```
Comment (pjvv) [inline crawler/helpers/cherrypicked_samples.py:19]: This could be done using set comprehension.

File: crawler/helpers/cherrypicked_samples.py [hunk 3]
```diff
@@ -0,0 +1,199 @@
...
+        samples (List[Sample]): List of samples in the shape of mongo documents
+        cherry_picked_samples (List[List[str]]): 2 dimensional list of cherry-picked samples with root sample id and
+        plate barcodes for each.
+
+    Returns:
+        List[Sample]: The original list of samples minus the cherry-picked samples.
+    """
+    cherry_picked_sets = [{cp_sample[0], cp_sample[1]} for cp_sample in cherry_picked_samples]
+    return list(
+        filter(
```
Comment (pjvv) [inline crawler/helpers/cherrypicked_samples.py:127]: First store the filter's results before converting to list to make it more readable?

File: crawler/helpers/cherrypicked_samples.py [hunk 4]
```diff
@@ -0,0 +1,199 @@
...
+def params_for_cherrypicked_samples_by_date_query(
+    root_sample_ids: List[str],
+    plate_barcodes: List[str],
+    start_date: str,
+    end_date: str,
+    chunk_size: int = 50000,
+) -> Iterator[Dict]:
+    for params in params_for_cherrypicked_samples_query(root_sample_ids, plate_barcodes, chunk_size):
+        yield {
+            **params,
```
Comment (pjvv) [inline crawler/helpers/cherrypicked_samples.py:155]: Please add a TODO here in the shape of:
```
TODO: Use [dictionary union](https://www.python.org/dev/peps/pep-0584/) when upgrading to Python 3.9
```

File: crawler/sql_queries.py [hunk 5]
```diff
@@ -120,3 +123,20 @@
 SQL_DART_SET_WELL_PROPERTY = "{CALL dbo.plDART_PlateUpdateWell (?,?,?,?)}"
 
 SQL_DART_ADD_PLATE = "{CALL dbo.plDART_PlateCreate (?,?,?)}"
+
+SQL_MLWH_GET_CP_SAMPLES = f"""
```
Comment (pjvv) [inline crawler/sql_queries.py:127]: See comment in other review.

File: crawler/sql_queries.py [hunk 6]
```diff
@@ -120,3 +123,20 @@
...
+    SELECT root_sample_id AS `{FIELD_ROOT_SAMPLE_ID}`, `{FIELD_PLATE_BARCODE}`,
+     phenotype AS `Result_lower`, `{FIELD_COORDINATE}`
+    FROM cherrypicked_samples
+    WHERE root_sample_id IN %(root_sample_ids)s
+    AND `{FIELD_PLATE_BARCODE}` IN %(plate_barcodes)s
+"""
+
+SQL_MLWH_GET_CP_SAMPLES_BY_DATE = f"""
```
Comment (pjvv) [inline crawler/sql_queries.py:135]: See comment in other review

File: setup_test_db.py [hunk 7]
```diff
@@ -333,6 +333,31 @@
 ) ENGINE=InnoDB AUTO_INCREMENT=17 DEFAULT CHARSET=utf8 COLLATE=utf8_unicode_ci;
 """
 
+create_cherrypicked_samples_view = """
```
Comment (pjvv) [inline setup_test_db.py:336]: See comment in other review

File: tests/helpers/test_cherrypicked_samples.py [hunk 8]
```diff
@@ -0,0 +1,461 @@
...
+    mock_cherry_picked_sample = [test_samples[0][FIELD_ROOT_SAMPLE_ID], test_samples[0][FIELD_PLATE_BARCODE]]
+
+    samples = remove_cherrypicked_samples(test_samples, [mock_cherry_picked_sample])
+    assert len(samples) == 7
+    assert mock_cherry_picked_sample[0] not in [sample[FIELD_ROOT_SAMPLE_ID] for sample in samples]
+
+
+# ----- get_cherrypicked_samples tests -----
+
+# Test Scenario
```
Comment (pjvv) [inline tests/helpers/test_cherrypicked_samples.py:110]: Moving these comments into the docstring make it easier to add and navigate in an IDE.

**Review by pjvv — CHANGES_REQUESTED** [review]

File: crawler/helpers/cherrypicked_samples.py (hunk 2 above)
Comment (harrietc52) [reply crawler/helpers/cherrypicked_samples.py:19]: We could do, but would it require looping over the samples list twice? Unless we're missing something

Comment (pjvv) [reply crawler/helpers/cherrypicked_samples.py:19]: Yeah, ignore my comment, you are building two distinct sets which would require you to loop over samples twice if using comprehension.

**Comment by harrietc52** [comment]
Test via SS: 
- [ ] Create file
- [ ] Upload to crawler
- [ ] Check sample in view 
- [ ] Cherrypick in SS. 
- [ ] Check sample in view and cherrypicked true

Test via

**Comment by emrojo** [comment]
Current request in UAT without using the view: 
image

**Review by harrietc52 — APPROVED** [review]

**Pull request merged**